itest/psbt_test.go

```diff
@@ -1608,6 +1619,436 @@ func testPsbtSighashNoneInvalid(t *harnessTest) {
+	bobVOut.AnchorOutputBip32Derivation = nil
+	bobVOut.AnchorOutputTaprootBip32Derivation = nil
+	bobVOut.SetAnchorInternalKey(
+		bobAnchorInternalKey, harnessNetParams.HDCoinType,
+	)
+	deliveryAddrStr := fmt.Sprintf(
+		"%s://%s", proof.UniverseRpcCourierType,
+		t.universeServer.ListenAddr,
+	)
+	deliveryAddr, err := url.Parse(deliveryAddrStr)
+	require.NoError(t.t, err)
+	bobVPsbt.Outputs[0].ProofDeliveryAddress = deliveryAddr
+
+	// The key information on the btc level, including the derivation path,
+	// needs to be updated to point to Bob's keys as well. Otherwise, he
+	// wouldn't be able to take over custody of the anchor carrying the assets.
+	btcpsbt.Outputs[1].TaprootInternalKey = schnorr.SerializePubKey(
+		bobAnchorInternalKey.PubKey,
+	)
+	btcpsbt.Outputs[1].Bip32Derivation = bobVOut.AnchorOutputBip32Derivation
+	btcpsbt.Outputs[1].TaprootBip32Derivation =
+		bobVOut.AnchorOutputTaprootBip32Derivation
+
+	// Before Bob tidies up the output commitments he keeps a backup of the
+	// transfer witnesses. This is where Alice's SIGHASH_NONE signature
+	// lies.
+	witnessBackup := bobVPsbt.Outputs[0].Asset.PrevWitnesses
+
+	// Bob tidies up the outputs.
+	err = tapsend.PrepareOutputAssets(ctxb, bobVPsbt)
+	require.NoError(t.t, err)
+
+	require.Len(t.t, bobVPsbt.Outputs, 1)
+	require.Equal(
+		t.t, bobVPsbt.Outputs[0].ScriptKey,
+		bobVPsbt.Outputs[0].Asset.ScriptKey,
+	)
+
+	// Bob restores Alice's signature for the asset input.
+	bobVPsbt.Outputs[0].Asset.PrevWitnesses = witnessBackup
+
+	bobVPsbtBytes, err := tappsbt.Encode(bobVPsbt)
+	require.NoError(t.t, err)
+
+	// Now let's serialize the edited vPSBT and commit it to our bitcoin
+	// PSBT.
+	b.Reset()
+	err = btcpsbt.Serialize(&b)
+	require.NoError(t.t, err)
+
+	// This call will also fund the PSBT, which means that the bitcoin that
+	// Alice "requested" previously by bumping her output will now be provided
+	// by Bob.
+	resp, err = bob.CommitVirtualPsbts(
+		ctxb, &wrpc.CommitVirtualPsbtsRequest{
+			VirtualPsbts: [][]byte{bobVPsbtBytes},
+			AnchorPsbt:   b.Bytes(),
+			AnchorChangeOutput: &wrpc.CommitVirtualPsbtsRequest_Add{
+				Add: true,
+			},
+			Fees: &wrpc.CommitVirtualPsbtsRequest_TargetConf{
+				TargetConf: 12,
+			},
+		},
+	)
+	require.NoError(t.t, err)
+
+	bobVPsbt, err = tappsbt.Decode(resp.VirtualPsbts[0])
+	require.NoError(t.t, err)
+
+	// Since Bob brings in a new input to the bitcoin transaction, he needs
+	// to sign it. We do not care about the sighash flag here, that can be
+	// the default, as we're not further editing the bitcoin transaction.
+	signResp := t.lndHarness.Bob.RPC.SignPsbt(
+		&walletrpc.SignPsbtRequest{
+			FundedPsbt: resp.AnchorPsbt,
+		},
+	)
+	require.NoError(t.t, err)
+
+	// Bob should sign exactly 1 input.
+	require.Len(t.t, signResp.SignedInputs, 1)
+
+	// The signed input should be at index 1, which is the bitcoin that Bob
+	// is bringing into the PSBT.
+	require.Equal(t.t, uint32(1), signResp.SignedInputs[0])
```

Comment:
Might be worth naming these indexes, something like: `bobPsbtInFundingIdx := uint32(1)`